src/overview/tabs/Details/cards/VmMigrationsHistoryChart.tsx

```diff
@@ -0,0 +1,166 @@
+import { useLayoutEffect, useRef, useState } from 'react';
+
+import {
+  Chart,
+  ChartArea,
+  ChartAxis,
+  ChartGroup,
+  ChartScatter,
+  ChartVoronoiContainer,
+} from '@patternfly/react-charts';
+import { getResizeObserver } from '@patternfly/react-core';
+import { useForkliftTranslation } from '@utils/i18n';
+
+import { ChartColors } from '../utils/colors';
+import type { MigrationDataPoint } from '../utils/toDataPointsHelper';
+import type { ChartDatumWithName } from '../utils/types';
+
+const VmMigrationsHistoryChart = ({
+  vmMigrationsDataPoints,
+}: {
+  vmMigrationsDataPoints: {
+    running: MigrationDataPoint[];
+    failed: MigrationDataPoint[];
+    succeeded: MigrationDataPoint[];
+  };
+}) => {
+  const { t } = useForkliftTranslation();
+  const chartContainerRef = useRef<HTMLDivElement>(null);
+  const [chartDimensions, setChartDimensions] = useState({ height: 400, width: 800 });
+
+  const handleResize = () => {
+    if (chartContainerRef.current?.clientHeight) {
+      const { clientHeight: height, clientWidth: width } = chartContainerRef.current;
+      setChartDimensions({ height, width });
+    }
+  };
+
+  useLayoutEffect(() => {
+    if (chartContainerRef.current) {
+      const resizeObserver = getResizeObserver(chartContainerRef.current, handleResize, true);
+      handleResize();
+
+      return () => {
+        resizeObserver();
+      };
+    }
+    return undefined;
+  }, []);
+
+  const maxVmMigrationValue = Math.max(
+    ...vmMigrationsDataPoints.running.map((migration) => migration.value),
+    ...vmMigrationsDataPoints.failed.map((migration) => migration.value),
+    ...vmMigrationsDataPoints.succeeded.map((migration) => migration.value),
+  );
+
+  const legendData = [
+    { name: t('Running'), symbol: { fill: ChartColors.Running } },
+    { name: t('Failed'), symbol: { fill: ChartColors.Failure } },
+    { name: t('Succeeded'), symbol: { fill: ChartColors.Success } },
+  ];
+
+  const maxTicks = Math.max(5, Math.ceil(maxVmMigrationValue) + 1);
+  const tickValues = Array.from({ length: maxTicks }, (_, i) => i + 1);
+
+  return (
+    <div ref={chartContainerRef} className="pf-u-h-100 pf-u-w-100">
+      <Chart
+        ariaDesc={t('Area chart with VM migration statistics')}
+        containerComponent={
+          <ChartVoronoiContainer
+            labels={({ datum }: { datum: ChartDatumWithName }) =>
+              datum.y === 0 || datum.name.includes('scatter')
+                ? (undefined as unknown as string)
```

Comment:
empty string isnt good here?

Reply:
empty string gets rendered and will mess up the legend box. `labels` prop expects a string, to get around that i do this casting so I can still pass `undefined`